enterprise/frontend/src/metabase-enterprise/data_editing/tables/edit/modals/EditingBaseRowModal.tsx

```diff
@@ -0,0 +1,103 @@
+import cx from "classnames";
+import { Fragment } from "react";
+import { t } from "ttag";
+
+import { FIELD_SEMANTIC_TYPES } from "metabase/lib/core";
+import {
+  ActionIcon,
+  Button,
+  Flex,
+  Group,
+  Icon,
+  type IconName,
+  Modal,
+  Text,
+  rem,
+} from "metabase/ui";
+import type { DatasetColumn, RowValues, Table } from "metabase-types/api";
+
+import { EditingBodyCellConditional } from "../inputs";
+
+import S from "./EditingBaseRowModal.module.css";
+
+interface EditingBaseRowModalProps {
+  datasetColumns: DatasetColumn[];
+  datasetTable?: Table;
+  onClose: () => void;
+  opened: boolean;
+  currentRowData?: RowValues;
+}
+
+const semanticIconMap: Record<string, IconName> = FIELD_SEMANTIC_TYPES.reduce(
```

Comment:
@deniskaber I couldn't find anything re-usable for column icons. Is this approach OK?